src/components/qas/QA.jsx

```diff
@@ -1,61 +1,34 @@
+/* eslint-disable react/jsx-props-no-spreading */
 import React from 'react';
 import PropTypes from 'prop-types';
 import QAExpansionList from './questions/qaExpansionList/index.jsx';
 import QATextInput from './questions/qaTextInput';
 import QASelect from './questions/qaSelect';
+import DistanceCalculator from './questions/qaCalculators/distanceCalculator';
 import Prompt from './questions/Prompt';
 
+const questions = {
+  accordion: QAExpansionList,
+  text: QATextInput,
+  textArea: QATextInput,
+  select: QASelect,
+  compoundSelect: QASelect,
+  'inline-select': QASelect,
+  DistanceCalculator,
+  prompt: Prompt,
+};
+
 class QA extends React.PureComponent {
   renderQA() {
-    const {
-      questionType,
-      question,
-      answer,
-      answerHandler,
-      editHandler,
-      saveHandler,
-      activeId,
-    } = this.props;
-
-    if (questionType === 'accordion') {
-      return (
-        <QAExpansionList
-          activeId={activeId}
-          question={question}
-          answer={answer}
-          cancelHandler={answerHandler}
-          saveHandler={saveHandler}
-          editHandler={editHandler}
-        />
-      );
-    }
+    const { questionType, answerHandler } = this.props;
 
-    if (questionType === 'text' || questionType === 'textArea') {
+    if (questionType) {
+      const QuestionComponent = questions[questionType];
       return (
-        <QATextInput
-          activeId={activeId}
-          question={question}
-          answer={answer}
-          answerHandler={answerHandler}
-        />
+        <QuestionComponent {...this.props} handleAnswerSelect={answerHandler} />
```

Comment:
This is such a clean, readable, great solution.  I'm clicking around the questions we have in there and I'm not seeing any problems, I assume you did the same.  I am honestly surprised and delighted this didn't precipitate more problems.  Great, great work.  I'm so glad this came up on Thursday.  Let's keep looking for new places to use this pattern.  I love where it gets us.

Reply:
I'm so happy that this worked as well as it did! I'll definitely keep my eye open for any other places we can use this pattern.